op-e2e/system_fpp_test.go

```diff
@@ -115,7 +103,19 @@ func testVerifyL2OutputRoot(t *testing.T, detached bool) {
 	fppConfig.L1URL = sys.NodeEndpoint("l1")
 	fppConfig.L2URL = sys.NodeEndpoint("sequencer")
 	fppConfig.DataDir = preimageDir
-	fppConfig.Detached = detached
+	if detached {
+		// When running in detached mode we need to compile the client executable since it will be called directly.
```

Comment:
I wonder eventually pull this out of `op-e2e` and have a CI pipeline which takes care of doing the building & then placing the result in the right place. Instead of a test binary that's run with `go test` it'd just be a binary that we'd have to run from CI.

The other idea I have here is building into a tmp directory & using that binary. This idea probably works better with the current setup.

Reply:
Yeah I was very much in two minds about the best way to do this. Part of the challenge currently is that most of our CI tasks have all their commands in the circleCI config and don't go through a makefile where it would be easy to run locally the same way and setup the right dependencies, env vars etc that more complex tests might require.

But I do quite like the way this just automatically works even when I just run it in GoLand.  I will extract the build command from the test itself to a helper function now I think about it though - then the test is clearer about what it wants to achieve and the details are better encapsulated if we want to change how it does it in the future (or if there are other build targets we want to execute they can all live together etc).